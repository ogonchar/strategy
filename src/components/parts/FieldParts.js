import React from 'react';
import { Field } from 'redux-form'

import * as CONST from "../../constants";


// Dropdown list linked directly to store 
const Select = ({ name, values, validate }) => {
    const options = () => {
        const res = values.map((opt, i) => {
            return <option key={i} value={opt}>{opt}</option>
        })
        return res
    }
    return (
        <div style = {wrapper}>
            <div style = {styleName}>{name}</div>
            <Field 
                name={name} 
                component='select'
                style = {dropSelect}
                validate = {validate}
            >
                {options()}
            </Field>
        </div>
    )
};

// Input linked directly to store 
const Input = ({ name, type, placeholder, validate }) => {
    return (
        <div style = {wrapper}>
            <div style = {styleName}>{name}</div>
            <Field
                name={name}
                component='input'
                type={type}
                placeholder={placeholder}
                style = {input}
                validate = {validate}
            />
        </div>
    )
}



const Button = ({ onClick, style, name }) => {
    return (
        <div
            onClick={onClick}
            style={{ ...buttonDiv, ...style }}
        >
            {name}
        </div>
    )
}


const Symbol = ({style, name, placeholder, type, validate }) => {
    return (
        <div style = {style}>
            <Field
                name={name}
                component='input'
                type={type}
                placeholder={placeholder}
                style = {{ ...style, ...symbolStyle}}
                validate = {validate}
            />
        </div>
    );
};

const Interval = ({ name, values, style, validate }) => {
    const options = () => {
        const res = values.map((opt, i) => {
            return <option key={i} value={opt}>{opt}</option>
        })
        return res
    }
    return (
        <div style = {wrapper}>
            <Field 
                name={name} 
                component='select'
                style = {{...dropSelect, ...style}}
                validate = {validate}
            >
                {options()}
            </Field>
        </div>
    )
};

export { Select, Input, Button, Symbol, Interval };

const common = {...CONST.HEIGHT, ...CONST.TEXT, 
    ...CONST.BACK, ...CONST.BORDER}


const wrapper = {
    display: 'flex',
    alignItems: 'center',
}

const buttonDiv = {
    ...common,
    cursor: 'pointer',
}

const styleName = {
    ...common,
    minWidth: 90,
}
const dropSelect = {
    ...common,
    minWidth: 102,
    height: (CONST.HEIGHT.height + 2)
  }
const input = {
    ...common,
    backgroundColor: '#f3f3f3ab',
    width: 100,
    minWidth: 100,
}
const symbolStyle = {

}