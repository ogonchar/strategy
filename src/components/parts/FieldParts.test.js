import { Button } from "./FieldParts";
import React from 'react';
import { mount } from 'enzyme';

test('show', () => {
    const wrapper = mount(
        <Button />
      );
      expect(wrapper).toMatchSnapshot()
})