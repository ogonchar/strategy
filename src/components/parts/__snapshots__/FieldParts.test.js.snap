// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <div
        style="height: 34px; text-align: left; font-size: 12px; display: flex; border: 1px solid #d2d2d2; border-radius: 3px; cursor: pointer;"
/>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "onClick": undefined,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#ccccccab",
          "border": "1px solid #d2d2d2",
          "borderRadius": "3px",
          "cursor": "pointer",
          "display": "flex",
          "fontSize": "12px",
          "height": 34,
          "justifyContent": "center",
          "textAlign": "left",
        },
      },
      "ref": null,
      "rendered": Array [
        undefined,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <div
          style="height: 34px; text-align: left; font-size: 12px; display: flex; border: 1px solid #d2d2d2; border-radius: 3px; cursor: pointer;"
/>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "onClick": undefined,
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "#ccccccab",
            "border": "1px solid #d2d2d2",
            "borderRadius": "3px",
            "cursor": "pointer",
            "display": "flex",
            "fontSize": "12px",
            "height": 34,
            "justifyContent": "center",
            "textAlign": "left",
          },
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
